Modix.Bot/Modules/EmojiStatsModule.cs

```diff
@@ -62,6 +63,48 @@ public async Task BottomGuildOnlyEmojiStatsAsync()
             await ReplyAsync(embed: embed.Build());
         }
 
+        [Command()]
+        [Priority(-10)]
+        [Summary("Gets the usage stats for a specific user.")]
+        public async Task UserEmojiStatsAsync(
+            [Summary("The user to retrieve stats for.")]
+                IUser user)
+        {
+            var userId = user.Id;
+            var guildId = Context.Guild.Id;
+
+            var emojiStats = await _emojiRepository.GetEmojiStatsAsync(guildId, SortDirection.Ascending, 10, userId: userId);
+            var userTotalUses = await _emojiRepository.SearchSummariesAsync(new EmojiSearchCriteria()
```

Comment:
I'm thinking we need a change on the repository level to support this and the above in one call rather than aggregating here.

Reply:
In that case, my suggestion is to wrap `EmojiUsageStatistics` into an object that carries `OldestTimestamp` and loosen the `recordLimit` parameter requirement. Does this sound good?

Reply:
Probably makes sense to add `OldestTimestamp` to the dto, yeah. Not sure what the benefit of loosening `recordLimit` is though. We still only want to pull no more than 10 records for the user, right?

Reply:
`var totalEmojiUsesPerDay = (double)userTotalUses.Count / numberOfDays;`

Can't show certain stats that depends on "All Time" usage otherwise

Reply:
Ohhhhh. I see. So maybe keep the dto the way it is and just make a separate repository method to get the footer level stuff like we do for the other commands.

Reply:
Using `EmojiUsageStatistics` would otherwise only show the total usage of your top 10 emojis

Reply:
Can you fill me in on why that approach would be better? (I genuinely don't know)

Reply:
Is aggregating directly in the db just faster?

Reply:
Faster, saves memory, less gc pressure, it's kinda what dbs are made to do

Reply:
Right, should I do a regular query for this and let you optimize it later with raw sql or do you want to just go straight for the sql variant?

Reply:
Would you rather pull 10,000 rows of data across the network, or 1? What about 100,000? What about 1,000,000?

Generally, any logic that can be performed by the database, should. The biggest exception is when EF can't properly translate the queries, which does occur rather often.

Reply:
How good are you at SQL, calle? :mmLul:

Reply:
I have a _very_ limited knowledge of raw SQL and would honestly rather leave it up to someone more skilled as I'm not entirely familiar with what is fast and efficient SQL-wise, my knowledge only go as far as knowing the syntax, essentially :p

Reply:
I hope I did this right :D